src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -62,6 +95,7 @@ public MemoryPoolIterator2 ProducingStart()
 
         public void ProducingComplete(MemoryPoolIterator2 end)
         {
+            // sync writes with is immedate == true as over Stream
```

Comment:
Then it would need to return a Task which would then change the `ISocketOutput` interface although... 1sec